src/Compilers/CSharp/Portable/Utilities/TypeSymbolExtensions.cs

```diff
@@ -252,8 +252,9 @@ private static NamedTypeSymbol GetDefaultBaseOrNull(NamedTypeSymbol type, CSharp
             {
                 case TypeKind.Class:
                 case TypeKind.Error:
-                case TypeKind.Interface:
                     return compilation.Assembly.GetSpecialType(SpecialType.System_Object);
+                case TypeKind.Interface:
+                    return null;
```

Comment:
>null [](start = 27, length = 4)

This would appear to change the behavior of `GetNextBaseTypeNoUseSiteDiagnostics` so that it is no longer consistent with `BaseType`.  Is this necessary to improve the diagnostic?  If so, why?

Reply:
> This would appear to change the behavior of GetNextBaseTypeNoUseSiteDiagnostics so that it is no longer consistent with BaseType.  Is this necessary to improve the diagnostic?  If so, why?

This code path affects only protected accessibility checks, and for the purpose of these checks an interface doesn't derive form an object. Without this check protected access to object members would be allowed since we now passing correct through type.

---
In reply to: [271078759](https://github.com/dotnet/roslyn/pull/34579#discussion_r271078759) [](ancestors = 271078759)

Reply:
Can you please change the name of this method and its callers up the stack including `GetNextBaseTypeNoUseSiteDiagnostics` to include something like `ForAccessChecks` and adjust the documentation of them to include saying that they are only for access checking?

Alternatively, do the filtering in the code that does the access checking.

Reply:
Aleksey explained to me that this change brings them (`BaseType` and `GetNextBaseTypeNoUseSiteDiagnostics`) into alignment (they were out of alignment before)

---
In reply to: [271085066](https://github.com/dotnet/roslyn/pull/34579#discussion_r271085066) [](ancestors = 271085066)